test/test_pullparser.rb

```diff
@@ -82,6 +82,28 @@ def test_character_references
       end
     end
 
+    def test_text_content_with_line_breaks
+      source = "<a>A</a><b>B\n</b><c>C\r\n</c>"
+      parser = REXML::Parsers::PullParser.new( source )
+      element_name = ''
+      while parser.has_next?
+        event = parser.pull
+        case event.event_type
+        when :start_element
+          element_name = event[0]
+        when :text
+          case element_name
+          when 'a'
+            assert_equal('A', event[1])
+          when 'b'
+            assert_equal("B\n", event[1])
+          when 'c'
+            assert_equal("C\n", event[1])
+          end
+        end
+      end
```

Comment:
This style may not find a bug because `assert_equal` is executed conditionally. (For example, if `:start_element` returns wrong element name, `case element_name; when ...` may be broken.)

How about using the following style?

```ruby
events = []
parser.each do |event|
  events << event
end
assert_equal([...], events) 
```

Reply:
(BTW, `parser.each.to_a` should work.)

Reply:
OK.
I see.
I will fix it.